lib/elixir/lib/map.ex

```diff
@@ -77,7 +77,10 @@ defmodule Map do
   """
   @spec new(Enum.t) :: map
   def new(enumerable) do
-    Enum.reduce(enumerable, %{}, fn {k, v}, acc -> put(acc, k, v) end)
+    enumerable
+    |> Enum.reduce([], &[&1 | &2])
+    |> :lists.reverse
```

Comment:
This is necessary because even though Maps are unordered, if we do `Map.new([a: 1, a: 2])` the current behaviour is that we get `%{a: 2}`. Without reversing we would get `%{a: 1}`